### PR TITLE
Add GitHub Actions competency

### PR DESCRIPTION
**Pull request by adarj**

### CODE REVIEW

File: competencies/github-actions.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Competency - GitHub Actions
+
+GitHub Actions is an API for cause and effect on GitHub: orchestrate any workflow, based on any event, while GitHub manages the execution, provides rich feedback, and secures every step along the way. With GitHub Actions, workflows and steps are just code in a repository, so you can create, share, reuse, and fork your software development practices.
+
+## How do you prove it?
+* You can create a continuous integration workflow for a project that automatically tests and deploys your code.
+* You can create a workflow that produces packages (such as NPM packages, Docker containers, etc.) and uploads them to GitHub Packages or another package hosting service.
```
Comment (codeallthethingz) [inline competencies/github-actions.md:7]: I'd remove docker containers as we probably want to build those in some other system.

File: competencies/github-actions.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Competency - GitHub Actions
+
+GitHub Actions is an API for cause and effect on GitHub: orchestrate any workflow, based on any event, while GitHub manages the execution, provides rich feedback, and secures every step along the way. With GitHub Actions, workflows and steps are just code in a repository, so you can create, share, reuse, and fork your software development practices.
+
+## How do you prove it?
+* You can create a continuous integration workflow for a project that automatically tests and deploys your code.
```
Comment (codeallthethingz) [inline competencies/github-actions.md:6]: I'd remove deploys from here.  Start with just a test.

File: competencies/github-actions.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+* You can create a workflow that produces packages (such as NPM packages, Docker containers, etc.) and uploads them to GitHub Packages or another package hosting service.
+* You know how to securely store secrets within GitHub, and are aware of the security risks in not doing so.
+
```
Comment (codeallthethingz) [inline competencies/github-actions.md:9]: Add some way to verify that they know about the different trigger mechanisms.

Comment (codeallthethingz) [inline competencies/github-actions.md:9]: Add something about resuing other peoples actions.

**Review by codeallthethingz — CHANGES_REQUESTED** [review]
Great start!  Some additions and changes

File: competencies/github-actions.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Competency - GitHub Actions
+
+GitHub Actions is an API for cause and effect on GitHub: orchestrate any workflow, based on any event, while GitHub manages the execution, provides rich feedback, and secures every step along the way. With GitHub Actions, workflows and steps are just code in a repository, so you can create, share, reuse, and fork your software development practices.
```
Comment (codeallthethingz) [inline competencies/github-actions.md:3]: are an API

**Comment by wafer-bw** [comment]
@adarj - The only thing I can suggest is maybe adding a blurb about using branch protection to require github action checks to pass before a PR can be merged. I'm not sure if that's fully relevant though... your call.

File: competencies/github-actions.md (hunk 3 above)
Comment (leukanos) [reply competencies/github-actions.md:9]: Would be also good to include:
- caching `actions/cache@v2`
- conditional steps 'if: github.event.pull_request.draft == false'
- artefacts `actions/upload-artifact@v2`
- steps vs jobs

File: competencies/github-actions.md (hunk 2 above)
Comment (adarj) [reply competencies/github-actions.md]: ✅

File: competencies/github-actions.md (hunk 4 above)
Comment (adarj) [reply competencies/github-actions.md]: ✅

File: competencies/github-actions.md (hunk 1 above)
Comment (adarj) [reply competencies/github-actions.md]: ✅

File: competencies/github-actions.md (hunk 3 above)
Comment (adarj) [reply competencies/github-actions.md:15]: ✅

Comment (adarj) [reply competencies/github-actions.md:16]: ✅

**Review by codeallthethingz — CHANGES_REQUESTED** [review]
see slack message for suggestions

**Review by ehacke — APPROVED** [review]
Looks good!

**Review by codeallthethingz — APPROVED** [review]
like it, merge it 🎉

**Pull request merged**